areas-of-d-usage.dd

```diff
@@ -0,0 +1,278 @@
+    which explains why D is a keystone language for their critical infrastructure.
+    eBay recently open-sourced their internal $(HTTPS github.com/eBay/tsv-utils-dlang, data processing utilities)
+    that are used in their a large-scale data mining environment.
+))
+
+$(SECTION2 $(LNAME2 industry, Industry), $(ARTICLE_FA_ICON institution)
+
+$(P Companies usually have codebases that are by far larger than regular
+    hobby projects, hence one of their major aims is to write perfect, elegant code
+    that is easy to understand and hence to maintain. D in comparison to other languages
+    like Java and C++, has many features that help to keep the codebase clean and
+    boilerplate-free - thus allowing one to work $(LINK2 $(ROOT_DIR)overview.html#productivity, productive).
+    For example have a look at D's simple and yet enormously powerful
+    $(LINK2 https://dlang.org/spec/template.html, templates)
+    or at D's $(LINK2 https://en.wikipedia.org/wiki/Uniform_Function_Call_Syntax, Uniform
+    Function Call Syntax) that allows any function to be called using the syntax
+    for method calls.
+)
+
+$(P Moreover $(LINK2 https://dlang.org/spec/function.html#pure-functions, pure functions),
+    the memory-safe subset $(LINK2 $(ROOT_DIR)safed.html, SafeD),
+    $(LINK2 $(ROOT_DIR)spec/unittest.html, special unittest blocks),
+    $(LINK2 $(ROOT_DIR)spec/contracts.html, contracts) and a sane approach to
+    $(LINK2 $(ROOT_DIR)spec/errors.html, error handling) help to
+    ship a bug-free program.
+    If at the end of the day $(LINK2 $(ROOT_DIR)overview.html#performance, performance)
+    matters - D has you covered!
+    You can tune your program with various tools starting from
+    $(LINK2 https://dlang.org/spec/version.html, conditional compilation) over
+    $(LINK2 https://dlang.org/spec/simd.html, SIMD vector optimizations) to
+    to $(LINK2 https://dlang.org/spec/iasm.html, Inline Assembler) and many more.
+)
+
+$(SECTION3 $(LNAME2 games, Games), $(ARTICLE_FA_ICON gamepad)
+
+    Binary compatibility to existing libraries is very important in large-scale
+    projects.
+    Remedy Entertainment has successfully shipped the first $(HTTP www.quantumbreak.com,
+    AAA game) to use D code for XBox One and Windows 10.
+    For more details see Ethan Watson's
+    $(HTTP dconf.org/2016/talks/watson.html, presentation).
+
+    There exist many pure D open source projects that show the rising popularity
+    of D among game developers.
+    For example have a look at the cross-platform 3D game engine
+    $(HTTP dash.circularstudios.com, Dash) or the D game development toolkit
+    $(HTTPS github.com/d-gamedev-team/gfm, gfm).
+)
+
+$(SECTION3 $(LNAME2 web, Web applications), $(ARTICLE_FA_ICON cloud)
+
+    D's ability to combine low and high-level code in one language, asynchronous
+    features like Fibers and compile-time function execution makes it well suited
```

Comment:
Well according to [N4024](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2014/n4024.pdf) we have both - `core.thread.Fiber` looks like `boost::coroutine` and **vibe.d**'s use of fibers is more like `boost::fiber`.

So I would say "stack-full coroutines and fibers".